lunaform/cluster.go

```diff
@@ -0,0 +1,269 @@
+// Copyright 2017 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package lunaform
+
+import (
+	"bufio"
+	"encoding/json"
+	"io/ioutil"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+
+	"github.com/coreos/mantle/harness"
+	"github.com/coreos/mantle/lunaform/tfdata"
+	"github.com/coreos/mantle/network"
+	"github.com/coreos/mantle/platform"
+	"github.com/coreos/mantle/platform/conf"
+)
+
+const gce = `
```

Comment:
@ethernetdan I don't know yet but more platforms and external terraform will need to be a later step.

@alexsomesan This is a new version of the test framework behind kola/pluton that uses terraform. The framework brings up a number of machines and executes a `go test` style test function. The current system creates machines via API calls (or running QEMU) but the hope is that using terraform will allow integration with the tectonic installer.